statsmodels/multivariate/manova.py

```diff
@@ -0,0 +1,176 @@
+    u = q.T.dot(y)
+    sscpr = np.subtract(y.T.dot(y), u.T.dot(u))
+    fittedvalues = u
+    return (df_resid, fittedvalues, sscpr)
+
+
+def _manova_test(hypotheses, fit_results, exog_names, endog_names):
+    def fn(L, M):
+        df_resid, u, sscpr = fit_results
+        t1 = L.dot(u)
+        q = matrix_rank(t1)
+        t1 = t1.dot(M)
+        H = t1.T.dot(t1)
+
+        # E = M'(Y'Y - B'(X'X)B)M
+        E = M.T.dot(sscpr).dot(M)
+        return E, H, q, df_resid
+
+    return _multivariate_test(hypotheses, exog_names, endog_names, fn)
+
+_manova_test.__doc__ = (
+    """
+    MANOVA hypotheses testing
+
+    For y = x * params, where y is dependent variables and x is independent
+    variables, testing L * params * M = 0 where L is the contast matrix for
+    hypotheses testing and M is the transformation matrix for transforming the
+    dependent variables in y.
+
+    Algorithm:
+        H = (L * u * M)' * (L * u * M)
+        E = M' * sscpr * M
+
+    .. [1] ftp://public.dhe.ibm.com/software/analytics/spss/documentation/statistics/20.0/en/client/Manuals/IBM_SPSS_Statistics_Algorithms.pdf
+
+    Parameters
+    ----------
+    """ + _hypotheses_doc +
+    """
+    fittedvalues : tuple
+        Output of ``fit_manova``
+    exog_names : a list of string
+    endog_names : a list of string
+
+    Returns
+    -------
+    a dict of manova tables
+    """)
+
+
+class MANOVA(Model):
+    """
+    Multivariate analysis of variance
+
+
+    Parameters
+    ----------
+    endog : array-like
+        Dependent variables (DV). A n_sample x n_y_var array where n_sample is
+        the number of observations and n_y_var is the number of DV.
+
+    exog : array-like
+        Independent variables (IV). A n_sample x n_x_var array where n is the
+        number of observations and n_x_var is the number of IV. An intercept is
+        not included by default and should be added by the user (models
+        specified using a formula include an intercept by default)
+
+    .. [1] ftp://public.dhe.ibm.com/software/analytics/spss/documentation/statistics/20.0/en/client/Manuals/IBM_SPSS_Statistics_Algorithms.pdf
+
+    Attributes
+    -----------
+    df_resid : float
+        The number of observation `n` minus the number of IV `q`.
+    sscpr : array
+        Sums of squares and cross-products of residuals
+    endog : array
+        See Parameters.
+    exog : array
+        See Parameters.
+    design_info : patsy.DesignInfo
+        Contain design info for the independent variables if model is
+        constructed using `from_formula`
+
+    """
+    def __init__(self, endog, exog, missing='none', hasconst=None, **kwargs):
+        super(MANOVA, self).__init__(endog, exog, **kwargs)
+        self.fittedmod = _manova_fit(self.exog, self.endog)
+
+    def mv_test(self, hypotheses=None):
+        if hypotheses is None:
+            if (hasattr(self, 'data') and self.data is not None and
+                        self.data.design_info is not None):
```

Comment:
I understand the purpose which is good,
but `design_info` has to be checked with `hasattr` instead of `is None`, AFAIU.
`self.data.design_info is not None` -> `hasattr(self.data, 'design_info')`

a quick check wit plain OLS
```
>>> mod_ols2 = OLS(mod_ols.endog, mod_ols.exog)
>>> mod_ols2.data.design_info
Traceback (most recent call last):
  File "<pyshell#100>", line 1, in <module>
    mod_ols2.data.design_info
AttributeError: 'ModelData' object has no attribute 'design_info'
```